### ISSUE TITLE
Feature request: Option to exclude dynamic imports

### ISSUE THREAD

**Issue by fongandrew** [sverweij/dependency-cruiser#141]
Now that dynamic import support is in, it'd be nice to have the option to not follow dynamic imports. Or at the very least, to exclude dynamic imports from a `no-circular` rule. AFAICT, there's no (documented) way to disable this though.

### Context
We use dependency-cruiser to make sure there aren't any circular dependencies with _static_ imports. We don't care so much about the dynamic ones though since they're less problematic than cycles affecting code that's run on module execution.

### Possible Solution
Perhaps a `noFollowDynamic` option?

**Comment by sverweij** [comment]
hi @fongandrew thanks for this suggestion - there's currently indeed no elegant way to distinguish dynamic imports in rules, and there should be. Especially when dependency-cruiser will start recognizing dynamic imports with arguments that can only be determined at runtime (which is [on the backlog](https://github.com/sverweij/dependency-cruiser/projects/1#card-21113518) )

They way I'm thinking to do this is to flag dynamic dependencies so they can be used in rules e.g. something like this:
```javascript
{
  forbidden: [
    {
      name: "no-circular",
      from: {
      },
      to {
        dynamicImport: false, //
        circular: true
      }
  ]
```

**Comment by fongandrew** [comment]
Ooh, that rule syntax would be perfect. Looking forward to it!

**Comment by sverweij** [comment]
hi @fongandrew - it's taken some time to address this issue; had to get #142 out the door first...

I've implemented my proposal above. It works, but it I realised it likely won't resolve the issue you have (see below) unless _all_ the dependencies in a circle are dynamic ones (see background below).

The `dynamic` attribute is probably going to stay because it's useful for other things, but I'm looking into a different solution for this issue (several ideas; including the followDynamic switch you proposed above - will ping you with one(s?) that works out).

### background below
Given this dependency-graph...
<img width="190" alt="sample" src="https://user-images.githubusercontent.com/4822597/59551197-0f78d200-8f76-11e9-9eb1-703656fc7ee0.png">

...with a rule that forbids circular dependencies, except for dynamic ones ...

```javascript
module.exports = { 
    forbidden: [{
            name: 'no-circular',
            severity: 'warn',
            from: {}, 
            to: {
                circular: true,
                dynamic: false
            }
    }]
}
```

... the result would be:
<img width="190" alt="sample" src="https://user-images.githubusercontent.com/4822597/59551222-6d0d1e80-8f76-11e9-9709-5247e0829958.png">

Close. No cigar ...

**Comment by fongandrew** [comment]
Oh, that's unfortunate, but I appreciate you taking the time to figure this out!

**Comment by sverweij** [comment]
hi @fongandrew so here's another stab, that works better in your use case - it excludes all dependencies that are dynamic:

```javascript
module.exports = {
  forbidden: [{
    name: 'no-circular',
    from: {},
    to: {
      circular: true
    }
  }],
  options: {
    exclude: {
      dynamic: true
    }
  }
}
```

This now works in dependency-cruiser@4.23.0-beta-1 on npm, so if you want to you can give it a spin - I'll be interested in your findings!

**Comment by fongandrew** [comment]
I tested it out. Works exactly as expected for our use case. Thank you very much!

**Comment by sverweij** [comment]
Awesome! I've merged the PR and published it as part of dependency-cruiser@4.23.0
